forcing_tools/netcdf2vic.py

```diff
@@ -0,0 +1,372 @@
+
+
+# -------------------------------------------------------------------- #
+def final(config_dict, ncfiles, points):
+
+    # ---------------------------------------------------------------- #
+    # close point files
+    for p in points:
+        p.close()
+    # ---------------------------------------------------------------- #
+
+    # ---------------------------------------------------------------- #
+    # report what was done
+    print('Finished netcdf2vic.py')
+    print('-->Wrote to {0} {1} '
+          'files'.format(len(points), config_dict['OPTIONS']['FILEFORMAT']))
+    print('-->Read from {0} netCDF files'.format(len(ncfiles)))
+    print('-->Output directory is here: '
+          '{0}'.format(config_dict['OPTIONS']['OUTPATH']))
+    # ---------------------------------------------------------------- #
+
+    return
+# -------------------------------------------------------------------- #
+
+
+# -------------------------------------------------------------------- #
+def process_command_line():
+    """
+    Parse arguments and assign flags for further loading of variables, for
+    information on input arguments, type netcdf2vic.py -h
+    """
+    # Parse arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument("config", type=str, help="Input Configuration File")
+    args = parser.parse_args()
+
+    config_dict = read_config(args.config)
+
+    return config_dict
+
+
+# -------------------------------------------------------------------- #
+# Read the Configuration File
+def read_config(config_file):
+    """
+    Return a dictionary with subdictionaries of all configFile options/values
+    """
+    config = ConfigParser.SafeConfigParser()
+    config.optionxform = str
+    config.read(config_file)
+    sections = config.sections()
+    dict1 = OrderedDict()
+    for section in sections:
+        options = config.options(section)
+        dict2 = OrderedDict()
+        for option in options:
+            dict2[option] = config_type(config.get(section, option))
+        dict1[section] = dict2
+    return dict1
+# -------------------------------------------------------------------- #
+
+
+# -------------------------------------------------------------------- #
+# Find the type of the config options
+def config_type(value):
+    """
+    Parse the type of the configuration file option.
+    First see the value is a bool, then try float, finally return a string.
+    """
+    val_list = [x.strip() for x in value.split(',')]
+    if len(val_list) == 1:
+        value = val_list[0]
+        if value in ['true', 'True', 'TRUE', 'T']:
+            return True
+        elif value in ['false', 'False', 'FALSE', 'F']:
+            return False
+        elif value in ['none', 'None', 'NONE', '']:
+            return None
+        elif isint(value):
+            return int(value)
+        elif isfloat(value):
+            return float(value)
+        else:
```

Comment:
I'm a bit confused as to what's going on here - need more comments, but it also seems clunky